codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -74,6 +77,9 @@ public HttpHeaders add(HttpHeaders headers) {
     @Override
     public HttpHeaders set(HttpHeaders headers) {
         if (headers instanceof DefaultHttpHeaders) {
+            if (headers == this) {
+                throw new IllegalArgumentException("can't add to itself.");
```

Comment:
In 4.1 we are not consistent in that this method `returns this;` and `add` throws an exception. I think we should be consistent in how we behave across versions (and how we handle this). So do we want to throw an exception, or just `return this;`?